routes/decoratorQueryArgs.go

```diff
@@ -61,11 +61,13 @@ type (
 	}
 
 	// QueryArg defines an argument by its name and its type. A description
-	// can be used for documentation purposes.
+	// can be used for documentation purposes. The argument can be optional
+	// by changing the Optional value to true.
 	QueryArg struct {
 		Name        string
 		Description string
 		Type        QueryParser
+		Optional    bool
```

Comment:
Decorators also generate documentation. You add this parameter yet you don’t change the documentation generation and optional arguments are still marked as required. Please edit the documentation generation logic to show optional arguments as optional.